### PR TITLE
fuzzier json number equatable

### PR DESCRIPTION
**Pull request by loganwright**
One potential implementation of fuzzier json number equality. When creating a double, I don't know how to check if below Double max similar to others if someone has suggestions.

### CODE REVIEW

File: Sources/JSON.swift [hunk 1]
```diff
@@ -12,13 +12,57 @@ public enum JSON {
...
+            } else {
+                return nil
+            }
+        }
+    }
+
+    private var uint: UInt? {
+        switch self {
+        case let .double(d) where d >= 0:
+            return UInt(d)
```
Comment (czechboy0) [inline Sources/JSON.swift:47]: Doesn't this cut the floating point part of the number?

```
let d = 2.1
UInt(d) //2
```

Comment (czechboy0) [inline Sources/JSON.swift:47]: So if you'd be comparing `UInt(2)` and `Double(2.1)`, it'd return `true`. That doesn't seem right

**Comment by loganwright** [comment]
@czechboy0 totally right, just updated!

**Comment by czechboy0** [comment]
Cool, just left one comment for something that I think should be fixed. Once that's done I'll merge, thanks @LoganWright! (Even though it's not 100% correct (Int.max gets cut when converted to Double), but 99.99% might be good enough here, it definitely is easier to use)

**Pull request merged**